lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java

```diff
@@ -59,42 +67,52 @@ static Dictionary loadDictionary(Path aff) throws IOException, ParseException {
   }
 
   public void testDictionariesLoadSuccessfully() throws Exception {
-    int failures = 0;
-    for (Path aff : findAllAffixFiles().collect(Collectors.toList())) {
-      try {
-        System.out.println(aff + "\t" + memoryUsage(loadDictionary(aff)));
-      } catch (Throwable e) {
-        failures++;
-        System.err.println("While checking " + aff + ":");
-        e.printStackTrace();
+    int threads = Runtime.getRuntime().availableProcessors();
+    ExecutorService executor = Executors.newFixedThreadPool(threads);
+    try {
+      Deque<Path> failures = new ConcurrentLinkedDeque<>();
```

Comment:
`synchronizedList` might look less surprising to readers :)

Reply:
Doesn't matter it a test, but sure.